blockchain/chainio_test.go

```diff
@@ -895,293 +831,158 @@ func TestUtxoSerialization(t *testing.T) {
-						compressed:    false,
-					},
-				},
-			},
-			serialized: hexToBytes("01df3982a731010132000596b538e853519c726a2c91e61ec11600ae1390813a627c66fb8be7947be63c52"),
-		},
-		{
-			name: "Only output 1, not coinbase",
-			entry: &UtxoEntry{
-				txVersion:  1,
-				isCoinBase: false,
-				hasExpiry:  false,
-				txType:     0,
-				height:     55555,
-				index:      1,
-				stakeExtra: nil,
-				sparseOutputs: map[uint32]*utxoOutput{
-					1: {
-						amount:        1000000,
-						scriptVersion: 0,
-						pkScript:      hexToBytes("76a914ee8bd501094a7d5ca318da2506de35e1cb025ddc88ac"),
-						compressed:    false,
-					},
-				},
-			},
-			serialized: hexToBytes("0182b103010002070000ee8bd501094a7d5ca318da2506de35e1cb025ddc"),
-		},
-		{
-			name: "Ticket with one output",
-			entry: &UtxoEntry{
-				txVersion:  1,
-				isCoinBase: false,
-				hasExpiry:  false,
-				txType:     1,
-				height:     55555,
-				index:      1,
-				stakeExtra: hexToBytes("030f001aba76a9140cdf9941c0c221243cb8672cd1ad2c4c0933850588ac0000206a1e1a221182c26bbae681e4d96d452794e1951e70a208520000000000000054b5f466001abd76a9146c4f8b15918566534d134be7d7004b7f481bf36988ac"),
-				sparseOutputs: map[uint32]*utxoOutput{
-					1: {
-						amount:        1000000,
-						scriptVersion: 0,
-						pkScript:      hexToBytes("76a914ee8bd501094a7d5ca318da2506de35e1cb025ddc88ac"),
-						compressed:    false,
-					},
-				},
-			},
-			serialized: hexToBytes("0182b103010402070000ee8bd501094a7d5ca318da2506de35e1cb025ddc030f001aba76a9140cdf9941c0c221243cb8672cd1ad2c4c0933850588ac0000206a1e1a221182c26bbae681e4d96d452794e1951e70a208520000000000000054b5f466001abd76a9146c4f8b15918566534d134be7d7004b7f481bf36988ac"),
-		},
-		{
-			name: "Only output 2, coinbase, non-zero script version",
+			name: "Coinbase, even uncomp pubkey",
 			entry: &UtxoEntry{
-				txVersion:  1,
-				isCoinBase: true,
-				hasExpiry:  false,
-				txType:     0,
-				height:     12345,
-				index:      1,
-				sparseOutputs: map[uint32]*utxoOutput{
-					2: {
-						amount:        100937281,
-						scriptVersion: 0xffff,
-						pkScript:      hexToBytes("76a914da33f77cee27c2a975ed5124d7e4f7f97513510188ac"),
-						compressed:    false,
-					},
-				},
+				amount: 5000000000,
+				pkScript: hexToBytes("410496b538e853519c726a2c91e61ec11600ae1390813a6" +
+					"27c66fb8be7947be63c52da7589379515d4e0a604f8141781e62294721166bf621" +
+					"e73a82cbf2342c858eeac"),
+				blockHeight:   12345,
+				blockIndex:    54321,
+				scriptVersion: 0,
+				packedFlags: encodeUtxoFlags(
+					true,  // coinbase
+					false, // unspent
+					false, // not modified
+					false, // no expiry
+					stake.TxTypeRegular,
+				),
 			},
-			serialized: hexToBytes("01df390101000182b095bf4182fe7f00da33f77cee27c2a975ed5124d7e4f7f975135101"),
-		},
-		{
-			name: "outputs 0 and 2 not coinbase",
+			serialized: hexToBytes("df3982a7310132000496b538e853519c726a2c91e61ec11" +
```

Comment:
I have manually verified all of these serializations in `TestUtxoSerialization` are accurate.